test/test_losses.py

```diff
@@ -265,3 +267,58 @@ def test_kl_div_loss_2d(self, input, target, expected):
     def test_kl_div_loss_2d_without_reduction(self, input, target, expected):
         actual = kornia.losses.kl_div_loss_2d(input, target, reduction='none')
         assert_allclose(actual, expected)
+
+
+class TestTotalVariation:
+    # Total variation of constant vectors is 0
+    @pytest.mark.parametrize('input,expected', [
+        (torch.ones(3, 4, 5), torch.zeros(())),
+        (2 * torch.ones(2, 3, 4, 5), torch.zeros(2)),
+    ])
+    def test_tv_on_constant(self, input, expected):
+        actual = kornia.losses.total_variation(input)
+        assert_allclose(actual, expected)
+
+    # Total variation for 3D tensors
+    @pytest.mark.parametrize('input', [
+        torch.rand(3, 4, 5),
+        torch.rand(1, 2, 3),
+    ])
+    def test_tv_on_3d(self, input):
+        actual = kornia.losses.total_variation(input)
+        input_np = input.numpy()
+        pixel_dif1 = input_np[:, 1:, :] - input_np[:, :-1, :]
+        pixel_dif2 = input_np[:, :, 1:] - input_np[:, :, :-1]
+        expected = torch.tensor(np.sum(np.abs(pixel_dif1)) + np.sum(np.abs(pixel_dif2)))
+        assert_allclose(actual, expected)
+
+    # Total variation for 4D tensors
+    @pytest.mark.parametrize('input', [
+        torch.rand(2, 3, 4, 5),
+        torch.rand(3, 1, 2, 3),
+    ])
+    def test_tv_on_4d(self, input):
+        actual = kornia.losses.total_variation(input)
+        input_np = input.numpy()
+        pixel_dif1 = input_np[:, :, 1:, :] - input_np[:, :, :-1, :]
+        pixel_dif2 = input_np[:, :, :, 1:] - input_np[:, :, :, :-1]
+        expected = torch.tensor(np.sum(np.abs(pixel_dif1), axis=(1, 2, 3)) + np.sum(np.abs(pixel_dif2), axis=(1, 2, 3)))
```

Comment:
The TotalVariation code does these ops in torch terms. This is a test, hence I am using numpy to compute the expected values. Using torch to compute the values expected of the torch function is slightly circular.

Reply:
I see your point, however it would be nice if the tests would have the least amount of dependencies as possible. Image numpy changes how it handles axis in the future, then the tests will fail when the code is correct. Of course, it is extremely unlikely that numpy changes how it handles axis in the future but you see my point. It is nice that the code  uses only torch, for consistency sake.

One way you could do these tests is by providing an input for which you know the expected output see [this example](https://github.com/arraiyopensource/kornia/blob/d97f76fe77ab19744a7def88e5dcefc71ddbae0e/test/color/test_hsv.py#L14). 

P.S: Thank you for your pacience, I am aware I am being a bit picky with these changes.

Reply:
The TF version use a double approach for the tests:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/image_ops_test.py#L4190

Reply:
A double approach would also be valid althought I dont know if we necessarily need/want it

Reply:
See also this comment https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/image_ops_test.py#L4284 and the follow-up handmade test.

Reply:
vote for avoid using numpy

Reply:
Removed numpy dependency